Userland/Libraries/LibJS/Runtime/Intl/AbstractOperations.cpp

```diff
@@ -385,6 +402,25 @@ LocaleResult resolve_locale(Vector<String> const& requested_locales, LocaleOptio
     return result;
 }
 
+// 9.2.12 CoerceOptionsToObject ( options ), https://tc39.es/ecma402/#sec-coerceoptionstoobject
+Value coerce_options_to_object(GlobalObject& global_object, Value options)
+{
+    auto& vm = global_object.vm();
+
+    // 1. If options is undefined, then
+    if (options.is_undefined()) {
+        // a. Return ! OrdinaryObjectCreate(null).
+        return Object::create(global_object, nullptr);
+    }
+
+    // 2. Return ? ToObject(options).
+    options = options.to_object(global_object);
+    if (vm.exception())
+        return {};
+
+    return options;
```

Comment:
For the suggestion above:

```suggestion
    return options.to_object(global_object);
```

(exception case will return a nullptr, and needs to be checked at the call site anyway)